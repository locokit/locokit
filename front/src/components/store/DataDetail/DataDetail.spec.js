/* eslint-disable @typescript-eslint/camelcase */
import { mount } from '@vue/test-utils'

import { COLUMN_TYPE } from '@locokit/lck-glossary'

import DataDetail from '@/components/store/DataDetail/DataDetail.vue'

const defaultWrapperParams = {
  mocks: {
    t: key => key,
    $t: key => key,
    $toast: {
      add: jest.fn(),
    },
  },
}

// Mock variables
const mockTableView2Definition = {
  id: 'table_view_2',
  table_id: 'table_1',
  columns: [
    {
      text: 'Description',
      id: 'string_2_column',
      settings: {},
      table_id: 'table_1',
      column_type_id: COLUMN_TYPE.STRING,
      editable: false,
      position: 1,
      transmitted: false,
      default: '',
      displayed: true,
      required: false,
      sort: 'DESC',
      table_column_id: '',
      table_view_id: 'table_view_2',
      style: {},
    },
    {
      text: 'Point',
      id: 'geo_point_column',
      settings: {},
      table_id: 'table_1',
      column_type_id: COLUMN_TYPE.GEOMETRY_POINT,
      editable: false,
      position: 2,
      transmitted: false,
      default: '',
      displayed: true,
      required: false,
      sort: 'DESC',
      table_column_id: '',
      table_view_id: 'table_view_2',
      style: {},
    },
  ],
}

const mockTableView1Definition = {
  id: 'table_view_1',
  table_id: 'table_1',
  columns: [
    {
      text: 'Name',
      id: 'string_1_column',
      settings: {},
      table_id: 'table_1',
      column_type_id: COLUMN_TYPE.STRING,
      editable: false,
      position: 1,
      transmitted: false,
      default: '',
      displayed: true,
      required: false,
      sort: 'DESC',
      table_column_id: '',
      table_view_id: 'table_view_1',
      style: {},
    },
    {
      text: 'Polygon',
      id: 'geo_polygon_column',
      settings: {
        map_sources: [{
          id: mockTableView2Definition.id,
          field: 'geo_point_column',
          popup: true,
          selectable: true,
        }],
      },
      table_id: 'table_1',
      column_type_id: COLUMN_TYPE.GEOMETRY_POLYGON,
      editable: true,
      position: 2,
      transmitted: false,
      default: '',
      displayed: true,
      required: false,
      sort: 'DESC',
      table_column_id: '',
      table_view_id: 'table_view_1',
      style: {},
    },
    // {
    //   text: 'Line string',
    //   id: 'geo_linestring_column',
    //   settings: {},
    //   table_id: 'table_1',
    //   column_type_id: COLUMN_TYPE.GEOMETRY_LINESTRING,
    //   editable: true,
    //   position: 3,
    //   transmitted: false,
    //   default: '',
    //   displayed: true,
    //   required: false,
    //   sort: 'DESC',
    //   table_column_id: '',
    //   table_view_id: 'table_view_1',
    //   style: {},
    // },
  ],
}

const mockTableView1Content = {
  data: [
    {
      id: 't1r1',
      text: 'Table 1 - Row 1',
      table_id: 'table_1',
      data: {
        string_1_column: 'My first row',
        geo_polygon_column: 'SRID=4326;POLYGON((1.4 45.75,2 45.6,1.9 45.3,1.4 45.75))',
      },
    },
    {
      id: 't1r2',
      text: 'Table 1 - Row 2',
      table_id: 'table_1',
      data: {
        string_1_column: 'My second row',
        geo_polygon_column: 'SRID=4326;POLYGON((1.2 45.75,2 45.6,1.9 45.3,1.4 45.75))',
      },
    },
  ],
}

const mockTableView2Content = {
  data: [
    {
      id: 't1r1',
      text: 'Table 1 - Row 1',
      table_id: 'table_1',
      data: {
        string_2_column: 'My first row bis',
        geo_point_column: 'SRID=4326;POINT(10.2 48.75)',
      },
    },
    {
      id: 't1r3',
      text: 'Table 1 - Row 3',
      table_id: 'table_1',
      data: {
        string_2_column: 'My third row bis',
        geo_point_column: 'SRID=4326;POINT(10.2 48.75)',
      },
    },
  ],
}

const definitionWithConditionalDisplay = {
  text: 'Testing conditional display',
  table_id: '0076b18b-5c85-4587-bff7-bba95f7361ca',
  columns: [
    {
      text: 'Text',
      id: 'e065323c-1151-447f-be0f-6d2728117b38',
      table_id: '0076b18b-5c85-4587-bff7-bba95f7361ca',
      column_type_id: COLUMN_TYPE.STRING,
      order: null,
      filter: null,
      displayed: true,
      editable: true,
    },
    {
      text: 'Boolean / Checkbox',
      id: 'e165323c-1151-447f-be0f-6d2728117b38',
      table_id: '0076b18b-5c85-4587-bff7-bba95f7361ca',
      column_type_id: COLUMN_TYPE.BOOLEAN,
      order: null,
      filter: null,
      displayed: true,
      editable: true,
    },
    {
      text: 'Conditional text if previous boolean is checked',
      id: '882326b8-b634-4ebb-aeb3-a2a2aa082c11',
      table_id: '0076b18b-5c85-4587-bff7-bba95f7361ca',
      column_type_id: COLUMN_TYPE.TEXT,
      order: null,
      filter: null,
      displayed: true,
      editable: true,
      display_conditions: [{
        field_id: 'e165323c-1151-447f-be0f-6d2728117b38',
        operator: '$eq',
        value: true,
      }],
    },
    {
      text: 'Conditional text if previous boolean is not checked',
      id: 'cf1ad5b7-54d0-46a2-9bd2-8b1ecf059c42',
      table_id: '0076b18b-5c85-4587-bff7-bba95f7361ca',
      column_type_id: COLUMN_TYPE.TEXT,
      order: null,
      filter: null,
      displayed: true,
      editable: true,
      display_conditions: [{
        field_id: 'e165323c-1151-447f-be0f-6d2728117b38',
        operator: '$ne', // here we use $ne, because if field is pristine, it could be null
        value: true,
      }],
    },
    {
      text: 'Status',
      id: 'b15bfe13-b18b-40ca-9637-1611c789b7c2',
      settings: {
        values: {
          first: {
            label: 'Done',
          },
          second: {
            label: 'Out of order',

          },
          third: {
            label: 'For other case (third), please specify in next field.',
          },
          fourth: {
            label: 'For other case (fourth), please specify in next field.',
          },
        },
      },
      table_id: '0076b18b-5c85-4587-bff7-bba95f7361ca',
      column_type_id: COLUMN_TYPE.SINGLE_SELECT,
      order: null,
      filter: null,
      displayed: true,
      editable: true,
    },
    {
      text: 'Conditional text if previous single select is third / fourth value',
      id: '87e6fd3e-d016-4509-bfe5-89796812cc17',
      table_id: '0076b18b-5c85-4587-bff7-bba95f7361ca',
      column_type_id: COLUMN_TYPE.TEXT,
      order: null,
      filter: null,
      displayed: true,
      editable: true,
      display_conditions: [{
        field_id: 'b15bfe13-b18b-40ca-9637-1611c789b7c2',
        operator: '$in',
        value: ['third', 'fourth'],
      }],
    },
  ],
}

jest.mock('@/components/ui/ColumnType/Geometry/Map.vue', () => ({
  render: () => '<lck-stub-map />',
}))

describe('DataDetail', () => {
  describe('Manage the geographic columns', () => {
    describe('Manage the secondary sources', () => {
      let wrapper, spyOnEmit

      beforeEach(() => {
        // Component initialization
        wrapper = mount(DataDetail, {
          propsData: {
            workspaceId: '1',
          },
          ...defaultWrapperParams,
          stubs: ['lck-map'],
        })
        spyOnEmit = jest.spyOn(wrapper.vm, '$emit')
      })

      it('Initialization with an editable field that needs secondary sources', async () => {
        // Define a new definition
        await wrapper.setProps({
          definition: mockTableView1Definition,
          row: mockTableView1Content.data[0],
        })
        // An event is emitted to ask the secondary sources
        expect(spyOnEmit).toHaveBeenCalledTimes(1)
        expect(spyOnEmit).toHaveBeenCalledWith('get-secondary-sources', [mockTableView2Definition.id])
        // The map column field isn't displayed as the secondary sources are not yet available
        expect(wrapper.find('lck-map-stub').exists()).toBe(false)

        // The secondary sources are now received
        await wrapper.setProps({
          secondarySources: {
            [mockTableView2Definition.id]: {
              definition: mockTableView2Definition,
              content: mockTableView2Content.data,
            },
          },
        })
        // The map column field is now displayed as the secondary sources are available
        expect(wrapper.find('lck-map-stub').exists()).toBe(true)
      })

      it('Initialization with a read only field that do not need secondary source', async () => {
        // Define a new definition
        await wrapper.setProps({
          definition: mockTableView2Definition,
          row: mockTableView2Content.data[0],
        })
        // No event is emitted to ask the secondary sources
        expect(spyOnEmit).not.toHaveBeenCalled()
        // The map column field is displayed
        expect(wrapper.find('lck-map-stub').exists()).toBe(true)
      })
    })
    describe('Get the valid geo resources for a column', () => {
      let wrapper

      beforeEach(() => {
        // Component initialization
        wrapper = mount(DataDetail, {
          propsData: {
            workspaceId: '1',
          },
          ...defaultWrapperParams,
          stubs: ['lck-map'],
        })
      })

      it('Which is editable and whose the needed secondary sources are available', async () => {
        await wrapper.setProps({
          secondarySources: {
            [mockTableView2Definition.id]: {
              definition: mockTableView2Definition,
              content: mockTableView2Content.data,
            },
          },
          definition: mockTableView1Definition,
          row: mockTableView1Content.data[0],
        })
        // Check map component resources
        const geoResources = wrapper.vm.getLckGeoResources(mockTableView1Definition.columns[1])
        expect(geoResources.length).toBe(2)
        const [secondaryResource, columnResource] = geoResources
        // Check column resource
        // Editable resource
        expect(columnResource.editableGeometryTypes.has(COLUMN_TYPE.GEOMETRY_POLYGON)).toBe(true)
        // Focus on this resource
        expect(columnResource.excludeFromBounds).toBe(false)
        // Customize it
        expect(columnResource.layers[0]).toMatchObject({
          paint: {
            'fill-color': '#01426B',
          },
        })
        // Check secondary resources
        // Not editable resources
        expect(secondaryResource.editableGeometryTypes.size).toBe(0)
        // Don't focus on these resources
        expect(secondaryResource.excludeFromBounds).toBe(true)
        // Some map settings can be passed
        expect(secondaryResource.popupMode).toBe('click')
        expect(secondaryResource.selectable).toBe(true)
        // Only keep the features related to a different row
        expect(secondaryResource.features.length).toBe(1)
        expect(secondaryResource.features[0].id).toBe('t1r3:geo_point_column')
      })
      it('Which is editable and whose the needed secondary sources are not available', async () => {
        await wrapper.setProps({
          definition: mockTableView1Definition,
          row: mockTableView1Content.data[0],
        })
        // Check map component resources
        const geoResources = wrapper.vm.getLckGeoResources(mockTableView1Definition.columns[1])
        expect(geoResources.length).toBe(1)
        const columnResource = geoResources[0]
        // Check column resource
        // Editable resource
        expect(columnResource.editableGeometryTypes.has(COLUMN_TYPE.GEOMETRY_POLYGON)).toBe(true)
        // Focus on this resource
        expect(columnResource.excludeFromBounds).toBe(false)
        // Customize it
        expect(columnResource.layers[0]).toMatchObject({
          paint: {
            'fill-color': '#01426B',
          },
        })
      })
      it('Which is not editable and which has not any secondary source', async () => {
        await wrapper.setProps({
          definition: mockTableView2Definition,
          row: mockTableView2Content.data[0],
        })
        // Check map component resources
        const geoResources = wrapper.vm.getLckGeoResources(mockTableView2Definition.columns[1])
        expect(geoResources.length).toBe(1)
        const columnResource = geoResources[0]
        // Check column resource
        // Read only resource
        expect(columnResource.editableGeometryTypes.size).toBe(0)
        // Focus on this resource
        expect(columnResource.excludeFromBounds).toBe(false)
        // Customize it
        expect(columnResource.layers[0]).toMatchObject({
          paint: {
            'circle-color': '#01426B',
          },
        })
      })
    })
  })

  describe('Manage conditional fields ($eq, $ne, $in)', () => {
    it('display / hide fields if an initial condition is respected', () => {
      expect.assertions(4)
      const wrapper = mount(DataDetail, {
        propsData: {
          workspaceId: '1',
          definition: definitionWithConditionalDisplay,
          row: {
            data: {},
          },
        },
        ...defaultWrapperParams,
      })

      expect(wrapper.html()).toMatchSnapshot()

      const wrapperText = wrapper.text()

      expect(wrapperText.indexOf('Conditional text if previous boolean is checked')).toBe(-1)
      expect(wrapperText.indexOf('Conditional text if previous boolean is not checked') > -1).toBe(true)
      expect(wrapperText.indexOf('Conditional text if previous single select is third / fourth value')).toBe(-1)
    })
    it('display / hide fields if a condition change in time', async () => {
      expect.assertions(11)
      const wrapper = mount(DataDetail, {
        propsData: {
          workspaceId: '1',
        },
        ...defaultWrapperParams,
      })
      await wrapper.setProps({
        definition: definitionWithConditionalDisplay,
        row: {
          data: {},
        },
      })

      let wrapperText = wrapper.text()

      expect(wrapperText.indexOf('Conditional text if previous boolean is checked')).toBe(-1)
      expect(wrapperText.indexOf('Conditional text if previous boolean is not checked') > -1).toBe(true)
      expect(wrapperText.indexOf('Conditional text if previous single select is third / fourth value')).toBe(-1)

      expect(wrapper.html()).toMatchSnapshot()

      await wrapper.find('input[type=checkbox]').trigger('click')

      expect(wrapper.vm.$props.row.data['e165323c-1151-447f-be0f-6d2728117b38']).toBe(true)

      expect(wrapper.html()).toMatchSnapshot()

      wrapperText = wrapper.text()

      expect(wrapperText.indexOf('Conditional text if previous boolean is checked') > -1).toBe(true)
      expect(wrapperText.indexOf('Conditional text if previous boolean is not checked')).toBe(-1)

      await wrapper.find('.p-dropdown.p-component.p-inputwrapper.p-dropdown-clearable').trigger('click')
      await wrapper.find('li:nth-of-type(3)').trigger('click')
      expect(wrapper.vm.$props.row.data['b15bfe13-b18b-40ca-9637-1611c789b7c2']).toBe('third')

      wrapperText = wrapper.text()
      expect(wrapperText.indexOf('Conditional text if previous single select is third / fourth value') > -1).toBe(true)

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
