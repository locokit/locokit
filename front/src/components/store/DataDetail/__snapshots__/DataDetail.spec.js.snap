// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataDetail Manage conditional fields ($eq, $ne, $in) display / hide fields if a condition change in time 1`] = `
<div class="p-fluid">
  <!---->
  <div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="e065323c-1151-447f-be0f-6d2728117b38" class="lck-color-primary untouched pristine pending">
          Text
        </label> <!----></div> <div class="form-field-editable untouched pristine pending"><input class="p-inputtext p-component" id="e065323c-1151-447f-be0f-6d2728117b38" rows="7"> <span class="cell-state"></span></div> <span class="untouched pristine pending"></span></span>
</div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><div class="p-checkbox p-component"><div class="p-hidden-accessible"><input type="checkbox" id="e165323c-1151-447f-be0f-6d2728117b38" value=""></div> <div role="checkbox" class="p-checkbox-box"><span class="p-checkbox-icon"></span></div>
</div> <label for="e165323c-1151-447f-be0f-6d2728117b38" class="lck-color-primary untouched pristine pending">
  Boolean / Checkbox
</label> <span class="cell-state"></span></div>
<div class="form-field-editable untouched pristine pending">
  <!---->
  <!---->
</div> <span class="untouched pristine pending"></span></span></div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="cf1ad5b7-54d0-46a2-9bd2-8b1ecf059c42" class="lck-color-primary untouched pristine pending">
          Conditional text if previous boolean is not checked
        </label> <!----></div> <div class="form-field-editable untouched pristine pending"><textarea class="p-inputtextarea p-inputtext p-component" id="cf1ad5b7-54d0-46a2-9bd2-8b1ecf059c42" rows="7"></textarea> <span class="cell-state"></span></div> <span class="untouched pristine pending"></span></span></div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="b15bfe13-b18b-40ca-9637-1611c789b7c2" class="lck-color-primary untouched pristine pending">
          Status
        </label> <!----></div> <div class="form-field-editable untouched pristine pending"><div class="p-dropdown p-component p-inputwrapper p-dropdown-clearable" id="b15bfe13-b18b-40ca-9637-1611c789b7c2"><div class="p-hidden-accessible"><input type="text" readonly="readonly" aria-haspopup="listbox"></div> <!----> <span class="p-dropdown-label p-inputtext p-placeholder"><span>
            components.datatable.placeholder
          </span></span>
  <!---->
  <div role="button" aria-haspopup="listbox" class="p-dropdown-trigger"><span class="p-dropdown-trigger-icon pi pi-chevron-down"></span></div>
  <transition-stub name="p-connected-overlay">
    <!---->
  </transition-stub>
</div> <span class="cell-state"></span></div> <span class="untouched pristine pending"></span></span></div>
</div>
`;

exports[`DataDetail Manage conditional fields ($eq, $ne, $in) display / hide fields if a condition change in time 2`] = `
<div class="p-fluid">
  <!---->
  <div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="e065323c-1151-447f-be0f-6d2728117b38" class="lck-color-primary untouched pristine pending">
          Text
        </label> <!----></div> <div class="form-field-editable untouched pristine pending"><input class="p-inputtext p-component" id="e065323c-1151-447f-be0f-6d2728117b38" rows="7"> <span class="cell-state"></span></div> <span class="untouched pristine pending"></span></span>
</div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><div class="p-checkbox p-component p-checkbox-checked p-checkbox-focused"><div class="p-hidden-accessible"><input type="checkbox" id="e165323c-1151-447f-be0f-6d2728117b38" value=""></div> <div role="checkbox" class="p-checkbox-box p-highlight p-focus" aria-checked="true"><span class="p-checkbox-icon pi pi-check"></span></div>
</div> <label for="e165323c-1151-447f-be0f-6d2728117b38" class="lck-color-primary untouched dirty pending changed">
  Boolean / Checkbox
</label> <span class="cell-state"></span></div>
<div class="form-field-editable untouched dirty pending changed">
  <!---->
  <!---->
</div> <span class="untouched dirty pending changed"></span></span></div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="882326b8-b634-4ebb-aeb3-a2a2aa082c11" class="lck-color-primary untouched pristine pending">
          Conditional text if previous boolean is checked
        </label> <!----></div> <div class="form-field-editable untouched pristine pending"><textarea class="p-inputtextarea p-inputtext p-component" id="882326b8-b634-4ebb-aeb3-a2a2aa082c11" rows="7"></textarea> <span class="cell-state"></span></div> <span class="untouched pristine pending"></span></span></div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="b15bfe13-b18b-40ca-9637-1611c789b7c2" class="lck-color-primary untouched pristine pending">
          Status
        </label> <!----></div> <div class="form-field-editable untouched pristine pending"><div class="p-dropdown p-component p-inputwrapper p-dropdown-clearable" id="b15bfe13-b18b-40ca-9637-1611c789b7c2"><div class="p-hidden-accessible"><input type="text" readonly="readonly" aria-haspopup="listbox"></div> <!----> <span class="p-dropdown-label p-inputtext p-placeholder"><span>
            components.datatable.placeholder
          </span></span>
  <!---->
  <div role="button" aria-haspopup="listbox" class="p-dropdown-trigger"><span class="p-dropdown-trigger-icon pi pi-chevron-down"></span></div>
  <transition-stub name="p-connected-overlay">
    <!---->
  </transition-stub>
</div> <span class="cell-state"></span></div> <span class="untouched pristine pending"></span></span></div>
</div>
`;

exports[`DataDetail Manage conditional fields ($eq, $ne, $in) display / hide fields if a condition change in time 3`] = `
<div class="p-fluid">
  <!---->
  <div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="e065323c-1151-447f-be0f-6d2728117b38" class="lck-color-primary untouched pristine">
          Text
        </label> <!----></div> <div class="form-field-editable untouched pristine"><input class="p-inputtext p-component" id="e065323c-1151-447f-be0f-6d2728117b38" rows="7"> <span class="cell-state"></span></div> <span class="untouched pristine"></span></span>
</div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><div class="p-checkbox p-component p-checkbox-checked"><div class="p-hidden-accessible"><input type="checkbox" id="e165323c-1151-447f-be0f-6d2728117b38" value=""></div> <div role="checkbox" class="p-checkbox-box p-highlight" aria-checked="true"><span class="p-checkbox-icon pi pi-check"></span></div>
</div> <label for="e165323c-1151-447f-be0f-6d2728117b38" class="lck-color-primary touched dirty changed">
  Boolean / Checkbox
</label> <span class="cell-state"></span></div>
<div class="form-field-editable touched dirty changed">
  <!---->
  <!---->
</div> <span class="touched dirty changed"></span></span></div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="882326b8-b634-4ebb-aeb3-a2a2aa082c11" class="lck-color-primary untouched pristine">
          Conditional text if previous boolean is checked
        </label> <!----></div> <div class="form-field-editable untouched pristine"><textarea class="p-inputtextarea p-inputtext p-component" id="882326b8-b634-4ebb-aeb3-a2a2aa082c11" rows="7"></textarea> <span class="cell-state"></span></div> <span class="untouched pristine"></span></span></div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="b15bfe13-b18b-40ca-9637-1611c789b7c2" class="lck-color-primary untouched dirty pending changed">
          Status
        </label> <!----></div> <div class="form-field-editable untouched dirty pending changed"><div class="p-dropdown p-component p-inputwrapper p-dropdown-clearable p-focus p-inputwrapper-filled p-inputwrapper-focus" id="b15bfe13-b18b-40ca-9637-1611c789b7c2"><div class="p-hidden-accessible"><input type="text" readonly="readonly" aria-haspopup="listbox" aria-expanded="true"></div> <!----> <span class="p-dropdown-label p-inputtext"><span class="lck-badge"><span class="lck-badge-content">For other case (third), please specify in next field.</span></span>
  </span> <i class="p-dropdown-clear-icon pi pi-times"></i>
  <div role="button" aria-haspopup="listbox" class="p-dropdown-trigger" aria-expanded="true"><span class="p-dropdown-trigger-icon pi pi-chevron-down"></span></div>
  <transition-stub name="p-connected-overlay">
    <div class="p-dropdown-panel p-component" style="visibility: visible; display: none; transform-origin: top; top: 0px; left: 0px;">
      <!---->
      <div class="p-dropdown-items-wrapper" style="max-height: 200px;">
        <ul role="listbox" class="p-dropdown-items">
          <li aria-label="Done" role="option" class="p-dropdown-item"><span class="lck-badge"><span class="lck-badge-content">Done</span></span></li>
          <li aria-label="Out of order" role="option" class="p-dropdown-item"><span class="lck-badge"><span class="lck-badge-content">Out of order</span></span></li>
          <li aria-label="For other case (third), please specify in next field." role="option" class="p-dropdown-item p-highlight" aria-selected="true"><span class="lck-badge"><span class="lck-badge-content">For other case (third), please specify in next field.</span></span></li>
          <li aria-label="For other case (fourth), please specify in next field." role="option" class="p-dropdown-item"><span class="lck-badge"><span class="lck-badge-content">For other case (fourth), please specify in next field.</span></span></li>
          <!---->
        </ul>
      </div>
    </div>
  </transition-stub>
</div> <span class="cell-state"></span></div> <span class="untouched dirty pending changed"></span></span></div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="87e6fd3e-d016-4509-bfe5-89796812cc17" class="lck-color-primary untouched pristine pending">
          Conditional text if previous single select is third / fourth value
        </label> <!----></div> <div class="form-field-editable untouched pristine pending"><textarea class="p-inputtextarea p-inputtext p-component" id="87e6fd3e-d016-4509-bfe5-89796812cc17" rows="7"></textarea> <span class="cell-state"></span></div> <span class="untouched pristine pending"></span></span></div>
</div>
`;

exports[`DataDetail Manage conditional fields ($eq, $ne, $in) display / hide fields if an initial condition is respected 1`] = `
<div class="p-fluid">
  <!---->
  <div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="e065323c-1151-447f-be0f-6d2728117b38" class="lck-color-primary untouched pristine">
          Text
        </label> <!----></div> <div class="form-field-editable untouched pristine"><input class="p-inputtext p-component" id="e065323c-1151-447f-be0f-6d2728117b38" rows="7"> <span class="cell-state"></span></div> <span class="untouched pristine"></span></span>
</div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><div class="p-checkbox p-component"><div class="p-hidden-accessible"><input type="checkbox" id="e165323c-1151-447f-be0f-6d2728117b38" value=""></div> <div role="checkbox" class="p-checkbox-box"><span class="p-checkbox-icon"></span></div>
</div> <label for="e165323c-1151-447f-be0f-6d2728117b38" class="lck-color-primary untouched pristine">
  Boolean / Checkbox
</label> <span class="cell-state"></span></div>
<div class="form-field-editable untouched pristine">
  <!---->
  <!---->
</div> <span class="untouched pristine"></span></span></div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="cf1ad5b7-54d0-46a2-9bd2-8b1ecf059c42" class="lck-color-primary untouched pristine">
          Conditional text if previous boolean is not checked
        </label> <!----></div> <div class="form-field-editable untouched pristine"><textarea class="p-inputtextarea p-inputtext p-component" id="cf1ad5b7-54d0-46a2-9bd2-8b1ecf059c42" rows="7"></textarea> <span class="cell-state"></span></div> <span class="untouched pristine"></span></span></div>
<div class="p-field"><span style="position: relative;"><div class="p-d-flex" style="position: relative;"><!----> <label for="b15bfe13-b18b-40ca-9637-1611c789b7c2" class="lck-color-primary untouched pristine">
          Status
        </label> <!----></div> <div class="form-field-editable untouched pristine"><div class="p-dropdown p-component p-inputwrapper p-dropdown-clearable" id="b15bfe13-b18b-40ca-9637-1611c789b7c2"><div class="p-hidden-accessible"><input type="text" readonly="readonly" aria-haspopup="listbox"></div> <!----> <span class="p-dropdown-label p-inputtext p-placeholder"><span>
            components.datatable.placeholder
          </span></span>
  <!---->
  <div role="button" aria-haspopup="listbox" class="p-dropdown-trigger"><span class="p-dropdown-trigger-icon pi pi-chevron-down"></span></div>
  <transition-stub name="p-connected-overlay">
    <!---->
  </transition-stub>
</div> <span class="cell-state"></span></div> <span class="untouched pristine"></span></span></div>
</div>
`;
